test/Interop/Cxx/class/Inputs/constructors.h

```diff
@@ -176,4 +176,16 @@ class Value {
   int x;
 };
 
+namespace SynthesizeDefaultCtorFactoryForCxxRefTy {
+struct __attribute__((swift_attr("import_reference")))
+__attribute__((swift_attr("retain:Retain")))
+__attribute__((swift_attr("release:Release"))) CxxRefTy {
+public:
```

Comment:
Couple other test cases I would love to see:
Very important:
* Add class with user-provided default ctor
* Add a test where we already have a static factory that is renamed to be an initialiser in Swift and verify that we do not import the default ctor in that case
* Add a test with a deleted member operator new so we do not import the default ctor in that case
Less important (could be follow-ups):
* Add a test case with a non-public ctor
* Make sure this feature works with immortal references
* Add a class that only has a ctor with default argument and verify that we do not import the initialiser